classes/Cart.php

```diff
@@ -3786,11 +3786,23 @@ public function checkDiscountValidity($obj, $discounts, $order_total, $products,
     }
 
     /**
-     * Return useful information about the cart.
+     * Return useful information about the cart for display purpose.
      *
      * @return array Cart details
      */
     public function getSummaryDetails($id_lang = null, $refresh = false)
+    {
+        $summary = $this->getUnalteredSummaryDetails($id_lang, $refresh);
```

Comment:
```suggestion
        $summary = $this->getUnalteredSummaryDetails($id_lang, (bool) $refresh);
```
You can't trust untyped parameter